src/components/NAlert/NAlert.vue

```diff
@@ -0,0 +1,408 @@
+         * 
+         * @author Rushabh Mulraj Shah
+         * @since 0.0.42
+         * @param {string} state The state property received as a prop
+         */
+        setIcon(state) {
+            if(state !== undefined) {
+                let selectedState = ICON_NAMES[state];
+                return selectedState !== undefined ? selectedState : ICON_NAMES[STATES.INFO]; 
+            } else {
+                return ICON_NAMES[STATES.INFO];
+            }
+        }
+    }
+}
+</script>
+
+<style lang="less" scoped>
+@import "./../../base/base.less";
+
+.n-alert {
+    box-sizing: border-box;
+    display: flex;
+    align-items: center;
+    padding: 1rem;
+    gap: 0.8rem;
+    min-height: 4.8rem;
+    cursor: default;
+    border-radius: 1.6rem;
+    border: 0.1rem solid @LabelColor;
+    color: @TypographyPrimaryColor;
+    font-size: @BaseFontSize + 2;
+    font-family: @PrimaryFont;
+
+    &:focus {
+        outline: none;
+    }
+
+    &.n-alert-full-width {
+        width: 100%;
+
+        .n-alert-label-text,
+        .n-alert-text {
+            font-size: @BaseFontSize + 6;
+        }
+    }
+
+    &.n-alert-link-button-container {
+        .n-alert-label-text,
+        .n-alert-text {
+            font-size: @BaseFontSize + 4;
+        }
+    }
+
+    &.n-alert-info {
+        background: @ColorPrimary20;
+        border: 0.1rem solid @ColorPrimary50;
+
+        .n-alert-button {
+            background: @ColorPrimary50;
+        }
+    }
+
+    &.n-alert-success {
+        background: @Bubbles;
+        border: 0.1rem solid @SuccessColor;
+    }
+
+    &.n-alert-warn {
+        background: @ColorFeedbackWarning20;
+        border: 0.1rem solid @ColorFeedbackWarning50;
+    }
+
+    &.n-alert-error {
+        background: @ColorFeedbackError20;
+        border: 0.1rem solid @ColorFeedbackError50;
+    }
+
+    &.n-alert-loader-container {
+        background: @ColorPrimary20 !important;
+        border: 0.1rem solid @ColorPrimary50;
+    }
+
+    &.n-alert-custom-width {
+        .n-alert-content {
+            column-gap: 0.2rem;
+        }
+    }
+
+    &.n-alert-xs {
+        .n-alert-content {
+            flex-direction: column !important;
+            row-gap: 0.8rem;
+        }
+    }
+
+    .n-alert-content {
+        display: flex;
+        flex-direction: row;
+        justify-content: space-between;
+        column-gap: 8rem;
+        width: 100%;
+        align-items: center;
+
+        &.n-alert-extended {
+            flex-direction: column;
+            align-items: unset;
+            justify-content: center;
+            row-gap: 1.2rem;
+
+            .n-alert-button-container {
+                display: flex;
+                justify-content: flex-end;
+            }
+        }
+    }
+
+    .n-alert-icon-text-wrapper {
+        display: flex;
+        align-items: center;
+        column-gap: 0.4rem;
+
+        .n-alert-icon {
+            height: 26.6px !important;
```

Comment:
Can we avoid using important here. Also please use rem instead of px

Reply:
I personally never use `!important` unless extremely important. I'll review the case next week and see if there are alternatives to this.